crates/ruff_python_formatter/src/other/arguments.rs

```diff
@@ -152,3 +199,349 @@ impl FormatNodeRule<Arguments> for FormatArguments {
         Ok(())
     }
 }
+
+struct CommentsAroundText<'a> {
+    text: &'static str,
+    comments: &'a [SourceComment],
+}
+
+impl Format<PyFormatContext<'_>> for CommentsAroundText<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        if self.comments.is_empty() {
+            text(self.text).fmt(f)
+        } else {
+            // There might be own line comments in trailing, but those are weird and we can kinda
+            // ignore them
+            // ```python
+            // def f42(
+            //     a,
+            //     # leading comment (own line)
+            //     / # first trailing comment (end-of-line)
+            //     # trailing own line comment
+            //     ,
+            //     c,
+            // ):
+            // ```
+            let (leading, trailing) = self.comments.split_at(
+                self.comments
+                    .partition_point(|comment| comment.line_position().is_own_line()),
+            );
+            write!(
+                f,
+                [
+                    leading_comments(leading),
+                    text(self.text),
+                    trailing_comments(trailing)
+                ]
+            )
+        }
+    }
+}
+
+/// `/` and `*` in a function signature
+///
+/// ```text
+/// def f(arg_a, /, arg_b, *, arg_c): pass
+///            ^ ^  ^    ^ ^  ^ slash preceding end
+///              ^  ^    ^ ^  ^ slash
+///                 ^    ^ ^  ^ slash following start
+///                      ^ ^  ^ star preceding end
+///                        ^  ^ star
+///                           ^ star following start
+/// ```
+#[derive(Debug)]
+pub(crate) struct ArgumentSeparators {
+    /// (preceding_end, slash, following_start)
+    pub(crate) slash: Option<(TextSize, TextRange, TextSize)>,
+    /// (preceding_end, star, following_start)
+    pub(crate) star: Option<(TextSize, TextRange, TextSize)>,
+}
+
+/// Finds slash and star in `f(a, /, b, *, c)`
+pub(crate) fn find_argument_separators(
+    contents: &str,
+    arguments: &Arguments,
+) -> ArgumentSeparators {
+    // We only compute preceding_end and token location here since following_start depends on the
+    // star location, but the star location depends on slash's position
+    let slash = if arguments.posonlyargs.is_empty() {
+        // If there are no positional only arguments the default handling below is wrong
+        None
+    } else if let Some(preceding_end) = arguments.posonlyargs.last().map(Ranged::end) {
```

Comment:
Can we remove this branch? The `else` branch returns `None` and the `else if` can never be `true` if `posonlyargs` is empty.

Reply:
good catch, this was remnant of the defaults list time